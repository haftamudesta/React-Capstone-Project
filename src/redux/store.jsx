import { configureStore } from '@reduxjs/toolkit';
import memesReducer from './memeSlice';

const store = configureStore({
  reducer: {
    memes: memesReducer,
  },
});
export default store;
