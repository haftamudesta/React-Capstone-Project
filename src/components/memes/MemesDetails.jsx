import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { fetchingMemesInfo } from '../../redux/memeSlice';
import classes from './MemesDetail.module.css';

const MemesDetails = () => {
  const { memesId } = useParams();
  const dispatch = useDispatch();
  const { memesData } = useSelector((state) => state.memes);
  useEffect(() => {
    dispatch(fetchingMemesInfo());
  }, [dispatch, memesId]);

  const [filterMemes] = memesData.filter((data) => data.id === memesId);
  if (!filterMemes || memesData.isLoading) {
    return <h1>Loading...</h1>;
  }
  return (
    <div className={classes.memes__card}>
      <div>
        <img src={filterMemes.url} alt="" className={classes.memes__image} />
      </div>
      <div className={classes.memes__content}>
        <h1>
          Name:
          {filterMemes.name}
        </h1>
        <div className={classes.memes__parametrs}>
          <p>
            Width:
            {filterMemes.width}
          </p>
          <p>
            Height:
            {filterMemes.height}
          </p>
          <p>
            Box Count:
            {filterMemes.box_count}
          </p>
        </div>
      </div>
    </div>
  );
};

export default MemesDetails;
