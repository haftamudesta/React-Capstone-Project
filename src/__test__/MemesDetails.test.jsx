import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MemesDetails from '../components/memes/MemesDetails';
import store from '../redux/store';

it('Test MemesContainer renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemesDetails />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
