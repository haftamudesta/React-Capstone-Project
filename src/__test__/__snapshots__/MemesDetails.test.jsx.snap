// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MemesContainer renders correctly 1`] = `
<h1>
  Loading...
</h1>
`;
